.github/workflows/proto.yml

```diff
@@ -0,0 +1,29 @@
+name: Protobuf CI
+on:
+  push:
+    branches:
+      - main
+  merge_group:
+  workflow_dispatch:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  buf:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-action@v1
+        with:
+          token: ${{ secrets.BUF_TOKEN }}
+          input: "proto"
+          breaking_against: "${{ github.event.repository.clone_url }}#format=git,ref=${{ github.event.pull_request.base.sha || github.event.before }},subdir=proto"
```

Comment:
I can't say I am certain this covers all the possible corner cases but looks okay at first glance.